core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -165,19 +174,46 @@ private void setTelemetryModules(ApplicationInsightsXmlConfiguration appConfigur
     }
 
     /**
-     * Setting an instrumentation key
-     * @param instrumentationKey The instrumentation key found in the configuration.
+     * Setting an instrumentation key:
+     * First we try the system property '-DAPPLICATION_INSIGHTS_IKEY=i_key'
+     * Next we will try the environment variable 'APPLICATION_INSIGHTS_IKEY',
+     * Next we will try to fetch the i-key from the ApplicationInsights.xml
+     * @param userConfiguration The configuration that was represents the user's configuration in ApplicationInsights.xml.
      * @param configuration The configuration class.
-     * @return True if succeeded.
      */
-    private boolean setInstrumentationKey(String instrumentationKey, TelemetryConfiguration configuration) {
+    private void setInstrumentationKey(ApplicationInsightsXmlConfiguration userConfiguration, TelemetryConfiguration configuration) {
         try {
-            configuration.setInstrumentationKey(instrumentationKey);
+            // First, try to find the i-key as a system property '-DAPPLICATION_INSIGHTS_IKEY=i_key'
```

Comment:
Regarding the comment it still refers to the previous implementation (i.e. `-DAPPLICATION_.INS..`).
Consider changing to 

> First, check whether an i-key was provided as a java system property (i.e. using the -D flag)